### PR TITLE
[IRGen] Strip marker protocols in a few more places

### PR DESCRIPTION
**Pull request by xedin**
- Strip marker protocols when forming generic metadata requests

   Without this might might end up with a single element protocol compositions
   which are invalid.

- Move marker protocol stripping from `mangleTypeSymbol` to `mangleTypeForFlatUniqueTypeRef`

  The original check introduced by https://github.com/apple/swift/pull/71855
  is too broad. For concrete metadata we call the runtime demangler so
  we need to strip off marker protocols when mangling that string and
  `mangleTypeForReflection` already does that.

Resolves: rdar://128667580

### CODE REVIEW

File: test/Interpreter/protocol_composition_with_markers.swift [hunk 1]
```diff
@@ -27,3 +27,27 @@ do {
...
+}
+
+do {
+  print(G<any (C & Sendable)>.self)
+  // CHECK: G<C>
+
+  class D<T> {
+  }
+
+  print((D<Int> & Sendable).self)
```
Comment (slavapestov) [inline test/Interpreter/protocol_composition_with_markers.swift:42]: You still need to test this from generic context, ie

```
func f<T>(_: T.Type) {
  print(D<T> & Sendable.self)
}
```

**Review by slavapestov — APPROVED** [review]

File: test/Interpreter/protocol_composition_with_markers.swift (hunk 1 above)
Comment (xedin) [reply test/Interpreter/protocol_composition_with_markers.swift:42]: Ah sure!

Comment (xedin) [reply test/Interpreter/protocol_composition_with_markers.swift:42]: ✅

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test

**Pull request merged**